db/migrate/20170110090935_add_ems_to_miq_alert_status.rb

```diff
@@ -0,0 +1,31 @@
+class AddEmsToMiqAlertStatus < ActiveRecord::Migration[5.0]
+  class ExtManagementSystem < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled # disable STI
+    has_many :vms, :class_name => "AddEmsToMiqAlertStatus::Vm"
+  end
+
+  class Vm < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled # disable STI
+    belongs_to :ext_management_system, :class_name  => "AddEmsToMiqAlertStatus::ExtManagementSystem",
+                                       :foreign_key => :ems_id
+    has_many :miq_alert_statuses, :class_name => "AddEmsToMiqAlertStatus::MiqAlertStatus", :as => :resource
+  end
+
+  class MiqAlertStatus < ActiveRecord::Base
+    belongs_to :resource, :polymorphic => true
+  end
+
+  def up
+    add_column :miq_alert_statuses, :ems_id, :bigint
+    say_with_time("add ems_id to miq alert statuses") do
+      MiqAlertStatus.all.each do |mas|
```

Comment:
What is the expected `MiqAlertStatus.count`?
How many different `resource_type` are we expecting?
This could be very slow.
Would it make sense to set these with an inner query or something?

```sql
update miq_alert_statuses
set ems_id = (select ems_id from vms where vm_id = resource_id)
where resource_type = "VmOrTemplate"
```

Not sure how many types we'd need. But the ruby would be something like.

```ruby
[Vms, Host].each do |model|
  # will have to update to arel :( - but it is a start
  MiqAlertStatus.where(:resource_type => model.name).update_all(:ems_id => Arel.sql("vm_id = resource_id"))
```